packages/wonder-blocks-icon-button/components/icon-button-core.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+import React from "react";
+import {StyleSheet} from "aphrodite";
+
+import Color, {mix, fade} from "wonder-blocks-color";
+import {addStyle} from "wonder-blocks-core";
+import type {ClickableHandlers} from "wonder-blocks-core";
+import type {SharedProps} from "./icon-button.js";
+
+type Props = SharedProps &
+    ClickableHandlers & {
+        hovered: boolean,
+        focused: boolean,
+        pressed: boolean,
+    };
+
+const StyledButton = addStyle("button");
+
+export default class IconButtonCore extends React.Component<Props> {
+    render() {
+        const {
+            icon,
+            alt,
+            color,
+            kind,
+            light,
+            disabled,
+            testId,
+            style,
+            hovered,
+            focused,
+            pressed,
+            ...handlers
+        } = this.props;
+
+        const buttonStyles = _generateStyles(color, kind, light);
+
+        const defaultStyle = [
+            sharedStyles.shared,
+            disabled && sharedStyles.disabled,
+            buttonStyles.default,
+            disabled && buttonStyles.disabled,
+            !disabled &&
+                (pressed
+                    ? buttonStyles.active
+                    : (hovered || focused) && buttonStyles.focus),
+        ];
+
+        // TODO: Replace the SVG with an Icon component once that's done
+        return (
+            <StyledButton
+                style={[sharedStyles.shared, defaultStyle, style]}
+                disabled={disabled}
+                data-test-id={testId}
+                {...handlers}
+            >
+                <svg
+                    role="img"
+                    aria-label={alt}
+                    width="20px"
+                    height="20px"
+                    viewBox="0 0 9.8 9.8"
+                >
+                    <title>{alt}</title>
```

Comment:
@somewhatabstract has been doing some work around labelling SVGs.  I believe the best practice is to set `aria-labelledby="title-id-123"` on the `svg` and then set `id="title-id-123"` on the `title` itself.

Reply:
Hm, should we taking in a separate `labelId` or something then? Or would the `Icon` component handle this?

Reply:
@chenyejia could you ask in the a11y channel if that alternative works.  It would be nice to avoid having to generate unique ids if possible.